apps/root/src/services/labelService.ts

```diff
@@ -32,19 +32,59 @@ export default class LabelService {
 
   async postLabels(labels: AccountLabels): Promise<void> {
     const user = this.accountService.getUser();
-    const currentLabels = this.labels;
     if (!user) {
       return;
     }
+    const currentLabels = this.labels;
     try {
       this.labels = { ...currentLabels, ...labels };
       await this.meanApiService.postAccountLabels(labels, user.id);
     } catch (e) {
-      this.labels = { ...currentLabels };
+      this.labels = currentLabels;
       console.error(e);
     }
   }
 
+  async editLabel(newLabel: string, labeledAddress: string): Promise<void> {
+    const user = this.accountService.getUser();
+    if (!user) {
+      return;
+    }
+    const currentLabels = this.labels;
+    if (!currentLabels.hasOwnProperty(labeledAddress)) {
+      console.warn(`Label for address ${labeledAddress} does not exist.`);
+      return;
+    }
+    try {
+      this.labels = { ...currentLabels, [labeledAddress]: newLabel };
+      await this.meanApiService.putAccountLabel(newLabel, labeledAddress, user.id);
+    } catch (e) {
+      this.labels = currentLabels;
+      console.error(e);
+    }
+  }
+
+  async deleteLabel(labeledAddress: string): Promise<void> {
+    const user = this.accountService.getUser();
+    if (!user) {
+      return;
+    }
+    const currentLabels = this.labels;
+    if (currentLabels.hasOwnProperty(labeledAddress)) {
+      const deletedLabel = this.labels[labeledAddress];
+      delete currentLabels[labeledAddress];
```

Comment:
should this actually be inside the try? since we want to revert all that when it fails right?

Reply:
You're right, it makes the code clearer to move those lines into the `try` block. Even though they're unlikely to fail due to the `hasOwnProperty` check, I think it's a good practice. I'll move them. Thanks for the feedback!